### PR TITLE
[SRVKE-1045] Adding the kafka controller to the HA replica setting

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

### CODE REVIEW

**Comment by matzew** [comment]
/assign @pierDipi 

as discussed starting to allow `controller` to be HA.
(default of kafka is `1` replica)

**Comment by matzew** [comment]
After we have replaced the `channel`, I'd like to move the default to `2` as well

File: knative-operator/pkg/controller/knativekafka/knativekafka_controller.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ var (
 	role              = mf.Any(mf.ByKind("ClusterRole"), mf.ByKind("Role"))
 	rolebinding       = mf.Any(mf.ByKind("ClusterRoleBinding"), mf.ByKind("RoleBinding"))
 	roleOrRoleBinding = mf.Any(role, rolebinding)
-	KafkaHAComponents = []string{"kafka-ch-controller"}
+	KafkaHAComponents = []string{"kafka-ch-controller", "kafka-controller"}
```
Comment (pierDipi) [inline knative-operator/pkg/controller/knativekafka/knativekafka_controller.go:55]: Should we add the webhook as well?

Comment (matzew) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: @pierDipi ah, yeah - I forgot about it.  Added it to the mix

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

File: knative-operator/pkg/controller/knativekafka/knativekafka_controller.go (hunk 1 above)
Comment (mgencur) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: Hey @pierDipi @matzew . Is "kafka-ch-controller" supposed to stay there in the list? This component is being removed in 1.23

Comment (pierDipi) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: I guess this PR isn't related to that, wasn't it removed on main?

Comment (pierDipi) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: removed from 1.23 as well, https://github.com/openshift-knative/serverless-operator/blob/73978d20a6dcbf9f780a39c9756edd7d5bbd61c2/knative-operator/pkg/controller/knativekafka/knativekafka_controller.go#L53

Comment (mgencur) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: Cool. Thanks!

**Pull request merged**